backend/vkt/src/main/java/fi/oph/vkt/service/PublicAuthService.java

```diff
@@ -19,55 +21,39 @@ public class PublicAuthService {
 
   private final CasTicketValidationService casTicketValidationService;
 
-  // TODO: authenticate person with information received from suomi.fi authentication service
-  @Transactional
-  public Person authenticate() {
-    final Random random = new Random();
-    final List<String> identityNumbers = List.of(
-      "200714-982U",
-      "010934-984D",
-      "210110-9320",
-      "230182-980D",
-      "130421-9046"
-    );
-    final String identityNumber = identityNumbers.get(random.nextInt(identityNumbers.size()));
-
-    return personRepository.findByIdentityNumber(identityNumber).orElseGet(() -> createPerson(identityNumber));
-  }
-
-  private Person createPerson(final String identityNumber) {
-    final Random random = new Random();
-    final List<String> lastNames = List.of("Aaltonen", "Alanen", "Eskola", "Hakala", "Heikkinen");
-    final List<String> firstNames = List.of("Anneli", "Ella", "Hanna", "Iiris", "Liisa");
-
+  private Person createPerson(final String identityNumber, final String firstName, final String lastName) {
     final Person person = new Person();
     person.setIdentityNumber(identityNumber);
-    person.setLastName(lastNames.get(random.nextInt(lastNames.size())));
-    person.setFirstName(firstNames.get(random.nextInt(firstNames.size())));
+    person.setLastName(lastName);
+    person.setFirstName(firstName);
 
     personRepository.saveAndFlush(person);
     return person;
   }
 
   @Transactional
-  public Person validate(final String ticket) {
-    final Random random = new Random();
-    final List<String> identityNumbers = List.of(
-      "200714-982U",
-      "010934-984D",
-      "210110-9320",
-      "230182-980D",
-      "130421-9046"
-    );
-
-    final boolean isValid = casTicketValidationService.validate(ticket);
+  public PublicPersonDTO validate(final String ticket) throws JsonProcessingException {
+    final CasResponse casResponse = casTicketValidationService.validate(ticket);
 
-    if (!isValid) {
+    if (casResponse.getAuthenticationSuccess().getUser().isEmpty()) {
       throw new APIException(APIExceptionType.INVALID_TICKET);
     }
 
-    final String identityNumber = identityNumbers.get(random.nextInt(identityNumbers.size()));
-
-    return personRepository.findByIdentityNumber(identityNumber).orElseGet(() -> createPerson(identityNumber));
+    final CasAttributes casAttributes = casResponse.getAuthenticationSuccess().getAttributes();
+    final String identityNumber = casAttributes.getNationalIdentificationNumber();
+    final String firstName = casAttributes.getFirstName();
+    final String lastName = casAttributes.getSn();
```

Comment:
Nää menee varmaankin oikein. Mietin vaan että mitäköhän tietoja mm. `givenName` ja `cn` ovat.

Reply:
Tsekkasin näitä:

cn (common name) on sukunimi + kaikki etunimet`<cas:cn>Testilä Tessa</cas:cn>`
firstName on itseasiassa kaikki etunimet. givenName antaa valitun etunimen arvon tai ensimmäisen etunimen, jos givenName ei ole annettu.

Katottiin dailyssa kans niin yki:ssa napataan kans `firstName` tietue cas-responsesta niin otetaan sama täälläki